airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/remove_fields.py

```diff
@@ -55,7 +60,11 @@ def transform(
         for pointer in self.field_pointers:
             # the dpath library by default doesn't delete fields from arrays
             try:
-                dpath.util.delete(record, pointer)
+                dpath.util.delete(
+                    record,
+                    pointer,
+                    afilter=(lambda x: self._filter_interpolator.eval(config or {}, property=x)) if self.condition else None,
```

Comment:
Is `afilter` a typo, should it be `filter`? I don't fully understand the details, so I might be wrong on this.

Reply:
that's just what the parameter is called in the dpath library https://pypi.org/project/dpath/